### PR TITLE
Added function for IPv6 support in turn.proxy.

### PR DESCRIPTION
**Pull request by Anisotrop**
Should be able to test STUN support for IPv6.

### CODE REVIEW

**Comment by Anisotrop** [comment]
@dpwspoon Hi David, I have created this PR to help QA check that IPv6 is supported by turn.proxy.

File: specification/turn/src/main/java/org/kaazing/specification/turn/internal/Functions.java [hunk 1]
```diff
@@ -67,6 +70,34 @@
     }
 
     @Function
+    public static byte[] ipV6XOR(String ip, byte[] transactionId) {
+        InetAddress address;
+        try {
+            address = InetAddress.getByName(ip);
+        } catch (UnknownHostException e) {
+            e.printStackTrace(); // TODO replace with log if available
```
Comment (dpwspoon) [inline specification/turn/src/main/java/org/kaazing/specification/turn/internal/Functions.java]: If you throw the exception (maybe Runtime), then the k3po runtime should throw the exception and print the stack trace... thoughts?

File: specification/turn/src/main/java/org/kaazing/specification/turn/internal/Functions.java [hunk 2]
```diff
@@ -67,6 +70,34 @@
...
+            return null;
+        }
+
+        if (!(address instanceof Inet6Address)) {
+            System.out.println("Address must be IPv6."); // TODO replace with log if available
```
Comment (dpwspoon) [inline specification/turn/src/main/java/org/kaazing/specification/turn/internal/Functions.java]: Same as above

**Comment by dpwspoon** [comment]
Relevant part of [Spec](https://tools.ietf.org/html/rfc5389#section-15.2):

>
 X-Port is computed by taking the mapped port in host byte order,
   XOR'ing it with the most significant 16 bits of the magic cookie, and
   then the converting the result to network byte order.  If the IP
   address family is IPv4, X-Address is computed by taking the mapped IP
   address in host byte order, XOR'ing it with the magic cookie, and
   converting the result to network byte order.  If the IP address
   family is IPv6, X-Address is computed by taking the mapped IP address
   in host byte order, XOR'ing it with the concatenation of the magic
   cookie and the 96-bit transaction ID, and converting the result to
   network byte order.

**Comment by Anisotrop** [comment]
@dpwspoon I have performed the changes, and execution can be traced to the function call when wrong parameters are given, but the error message does not seem to appear in the logs.

**Comment by dpwspoon** [comment]
Thanks @Anisotrop, I've filed the ticket to track that bug here: https://github.com/k3po/k3po/issues/358.  Please review to see if the details I got were correct,  I added question to label for now as I'd like you to confirm the observed behavior.

**Pull request merged**